src/rebuild.ts

```diff
@@ -53,22 +53,30 @@ class Rebuilder {
     this.lifecycle.emit('start');
 
     const rootPackageJson = await readPackageJson(this.buildPath);
-    const markWaiters: Promise<void>[] = [];
-    const depKeys = [];
+    const markWaiters: Promise<void>[] = this.types.reduce((acc: string[], curr: string) => {
```

Comment:
I personally don't like this `reduce` change, sure it's nice and reduce is a pretty cool ES6 feature but in this case I think the 3 consecutive if statements where much easier to understand than a reduce call